### ISSUE TITLE
Beta 9 reports "Missing Analyzer Reference"

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#1345]
When 1.0.0-beta009 is installed in a project, Visual Studio reports a MissingAnalyzerReference diagnostic:

> Analyzer assembly '{path}\StyleCopAnalyzers\packages\StyleCop.Analyzers.1.0.0-beta009\analyzers\dotnet\cs\Newtonsoft.Json.dll' depends on 'mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e' but it was not found. Analyzers may not run correctly.

**Comment by sharwell** [comment]
This issue is reported only inside the IDE. The AppVeyor build is still showing up clean with Beta 9.

**Comment by otac0n** [comment]
Removing the analyzer reference to Newtonsoft (Expand References\Analyzers, Right-click Newtonsoft.Json, Remove) appears to fix the warning.

I did this, and eventually I started seeing this *Info Message* showing up in VS:

    Message	AD0001	The User Diagnostic Analyzer 'StyleCop.Analyzers.DocumentationRules.FileHeaderAnalyzers' threw an exception of type 'System.IO.FileNotFoundException' with message 'Could not load file or assembly 'Newtonsoft.Json, Version=7.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. The system cannot find the file specified.'.	

    'StyleCop.Analyzers.DocumentationRules.FileHeaderAnalyzers' threw the following exception:
    'System.IO.FileNotFoundException: Could not load file or assembly 'Newtonsoft.Json, Version=7.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. The system cannot find the file specified.
    File name: 'Newtonsoft.Json, Version=7.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
       at StyleCop.Analyzers.SettingsHelper.GetStyleCopSettings(ImmutableArray`1 additionalFiles)
       at StyleCop.Analyzers.SettingsHelper.GetStyleCopSettings(SyntaxTreeAnalysisContext context)
       at StyleCop.Analyzers.DocumentationRules.FileHeaderAnalyzers.HandleSyntaxTreeAxtion(SyntaxTreeAnalysisContext context, Compilation compilation)
       at StyleCop.Analyzers.DocumentationRules.FileHeaderAnalyzers.<>c__DisplayClass80_0.<HandleCompilationStart>b__0(SyntaxTreeAnalysisContext ctx)
       at StyleCop.Analyzers.AnalyzerExtensions.<>c__DisplayClass1_0.<RegisterSyntaxTreeActionHonorExclusions>b__0(SyntaxTreeAnalysisContext c)
       at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__DisplayClass33_0.<ExecuteSyntaxTreeActions>b__0()
       at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock(DiagnosticAnalyzer analyzer, Action analyze)

    WRN: Assembly binding logging is turned OFF.
    To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
    Note: There is some performance penalty associated with assembly bind failure logging.
    To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
    '.

**Comment by jimmymain** [comment]
not sure if this will help: http://forum.linqpad.net/discussion/856/compile-errors-for-assemblies-which-reference-to-mscorlib-2-0-5-0-linqpad-5-00-08

**Comment by jimmymain** [comment]
I have obtained the source, and tried to take a look at this problem.
In attempting to build the solution, I have this same warning.

```
Severity	Code	Description	Project	File	Line
Warning	MissingAnalyzerReference	Analyzer assembly 'D:\StyleCopAnalyzers\packages\StyleCop.Analyzers.1.0.0-beta009\analyzers\dotnet\cs\Newtonsoft.Json.dll' 
depends on 'mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e' but it was not found. Analyzers may not run correctly.	StyleCop.Analyzers		1
```

Additionally, I have 9 errors, each of which is:
```
Severity	Code	Description	Project	File	Line
Error		Build action 'EmbeddedResource' is not supported by 
one or more of the project's targets.	
StyleCop.Analyzers	D:\StyleCopAnalyzers\StyleCop.Analyzers\StyleCop.Analyzers\OrderingRules\OrderingResources.resx	0
```

newtonsoft definitely has a dependency on mscorlib version 2.0.5.0, and is required by the solution.

**Comment by jimmymain** [comment]
I think this warning can be safely ignored.
There was a similar issue with the FxCopAnalyzer https://github.com/dotnet/roslyn/pull/3047/files

it was solved by white listing all the system dll's. hopefully the warnings can be suppressed. If not, it's possibly a bug in roslyn, mscorlib should have been white listed in addition to the system libraries.

An alternative, is to rebuild newtonsoft without the silverlight targets and reference that, since it would appear that the reference to 2.0.5.0 is for silverlight, and is not actually required.

**Comment by jimmymain** [comment]
referencing the net45 version of newtonsoft.json.dll instead of the portable version, for  both 6.0.8 and 7.0.1 of newtonsoft.json.dll solves the problem.

**Comment by NikolayIT** [comment]
Cleaning the solution fixed this for me.

**Comment by alxyr** [comment]
I just want to confirm that using net45 version of Newtonsoft.Json.dll indeed resolves the issue with the missing references to 'mscorlib, Version=2.0.5.0'. Would be nice not to see all those warnings in v1.0.0.10.

**Comment by sharwell** [comment]
Blocked on dotnet/roslyn#5065.

**Comment by nbarbettini** [comment]
Ditto, can confirm that copying the net45 version of JSON.NET 7.0.1 into `packages\StyleCop.Analyzers.1.0.0-beta010\analyzers\dotnet\cs` makes the IDE warnings go away.

**Comment by jimmymain** [comment]
It's definitely worth considering deploying the net45 version of JSON until microsoft have completed [roslyn/#5065](https://github.com/dotnet/roslyn/issues/5065)

**Comment by sharwell** [comment]
@jimmymain I think that could be a reasonable workaround. We could actually do it directly in the **.nuspec** file and not need to touch the projects at all.
